torchvision/utils.py

```diff
@@ -288,18 +289,17 @@ def draw_segmentation_masks(
     for color in colors:
         if isinstance(color, str):
             color = ImageColor.getrgb(color)
-        color = torch.tensor(color, dtype=out_dtype)
-        colors_.append(color)
+        colors_.append(torch.tensor(color, dtype=out_dtype))
 
     img_to_draw = image.detach().clone()
     # TODO: There might be a way to vectorize this
-    for mask, color in zip(masks, colors_):
-        img_to_draw[:, mask] = color[:, None]
+    for mask, color_tensor in zip(masks, colors_):
+        img_to_draw[:, mask] = color_tensor[:, None]
```

Comment:
In order to preserve git blame I would suggest to revert these changes, as I don't think they significantly improve readability either

Reply:
Problem is mypy. To satisfy mypy I had to change the name.

Reply:
what is mypy complaining about?

Reply:
Mypy complains that `color` isn't a tensor, as we are re-using the name color.
Here is the exact error.

```
torchvision/utils.py:292: error: Incompatible types in assignment (expression
has type "Tensor", variable has type "Union[str, Tuple[int, int, int]]") 
[assignment]
            color = torch.tensor(color, dtype=out_dtype)
                    ^
torchvision/utils.py:298: error: Invalid index type "Tuple[slice, None]" for
"Union[str, Tuple[int, int, int]]"; expected type "Union[int, slice]"  [index]
            img_to_draw[:, mask] = color[:, None]
                                   ^
torchvision/utils.py:298: error: Invalid tuple index type (actual type
"Tuple[slice, None]", expected type "Union[int, slice]")  [misc]
            img_to_draw[:, mask] = color[:, None]
                                   ^
torchvision/utils.py:298: error: Incompatible types in assignment (expression
has type "Union[str, Any]", target has type
"Union[Tensor, Union[int, float, bool]]")  [assignment]
            img_to_draw[:, mask] = color[:, None]
                                   ^
Found 4 errors in 1 file (checked 127 source files)

Exited with code exit status 1

```

The easiest fix was to rename, rather than ignoring mypy errors.

Reply:
Ok, let's leave it as-is then.

@pmeier what is your take on this?

Reply:
Given that this is one of the most frustrating things when working with `mypy`, we could simply put `allow_redefinition = True` in our `mypy` config. I also proposed doing this in #4513, albeit only for the prototype datasets.

Reply:
done https://github.com/pytorch/vision/pull/4531/files